xdem/spatial_tools.py

```diff
@@ -398,6 +408,149 @@ def hillshade(dem: Union[np.ndarray, np.ma.masked_array], resolution: Union[floa
     # The output is scaled by "(x + 0.6) / 1.84" to make it more similar to GDAL.
     return np.clip(255 * (shaded + 0.6) / 1.84, 0, 255).astype("float32")
 
+def get_xy_rotated(raster: gu.georaster.Raster, myang: float) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Rotate x, y axes of image to get along- and cross-track distances.
+    :param raster: raster to get x,y positions from.
+    :param myang: angle by which to rotate axes (degrees)
+    :returns xxr, yyr: arrays corresponding to along (x) and cross (y) track distances.
+    """
+    # creates matrices for along and across track distances from a reference dem and a raster angle map (in radians)
+
+    myang = np.deg2rad(myang)
+
+    # get grid coordinates
+    xx, yy = raster.coords(grid=True)
+    xx = xx - np.min(xx)
+    yy = yy - np.min(yy)
+
+    # get rotated coordinates
+    xxr = np.multiply(xx, np.cos(myang)) + np.multiply(-1 * yy, np.sin(myang))
+    yyr = np.multiply(xx, np.sin(myang)) + np.multiply(yy, np.cos(myang))
+
+    # re-initialize coordinate at zero
+    xxr = xxr - np.nanmin(xxr)
+    yyr = yyr - np.nanmin(yyr)
+
+    return xxr, yyr
+
+def choice_best_polynomial(cost: np.ndarray, margin_improvement : float = 20., verbose: bool = False) -> int:
```

Comment:
The use-case of this is very specific to `robust_polynomial_fit()`, right? If so, perhaps it should be prefixed with an underscore, as it may not be useful for the public interface.

Reply:
Good idea! Added